chunky/src/java/se/llbit/chunky/resources/Image.java

```diff
@@ -0,0 +1,31 @@
+package se.llbit.chunky.resources;
+
+/**
+ * Generic image giving read only access
+ */
+public interface Image {
+  /**
+   * Get the width of the image
+   */
+  int getWidth();
+
+  /**
+   * Get the height of the image
+   */
+  int getHeight();
+
+  /**
+   * Get the pixel at position (x, y)
+   */
+  int getPixel(int x, int y);
+
+  /**
+   * Get the pixel at the given index. Index is computed as y * width + x
+   */
+  int getPixel(int index);
+
+  /**
+   * Convert the image to a BitmapImage
+   */
+  BitmapImage asBitmap();
```

Comment:
`@PluginApi`?

Reply:
I did not really think of this interface as being used/useful for plugin

Reply:
The OpenCL plugin needs to get the textures somehow. I don't think any other plugin needs it so maybe we could not make it official api. :-)

Reply:
Oh, in this case it should probably rather be Texture.getBitmap that is exposed